hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAdmin2.java

```diff
@@ -864,4 +865,42 @@ public void testSlowLogResponses() throws Exception {
     Assert.assertEquals(onlineLogRecords.size(), 0);
   }
 
+  @Test
+  public void testGetRegionServers() throws Exception {
+    // get all live server names
+    List<ServerName> serverNames = new ArrayList<>(ADMIN.getRegionServers(true));
+    Assert.assertEquals(3, serverNames.size());
+
+    List<ServerName> serversToDecom = new ArrayList<>();
+    ServerName serverToDecommission = serverNames.get(0);
+
+    serversToDecom.add(serverToDecommission);
+    ADMIN.decommissionRegionServers(serversToDecom, false);
+    waitForServerDecom(serverToDecommission, true);
+
+    Assert.assertEquals(2, ADMIN.getRegionServers(true).size());
+
+    ADMIN.recommissionRegionServer(serverToDecommission, Collections.emptyList());
+
+    Assert.assertEquals(3, ADMIN.getRegionServers(true).size());
+    waitForServerDecom(serverToDecommission, false);
```

Comment:
Why decommission this server in the end?

Reply:
I was thinking the assertion `Assert.assertEquals(3, ADMIN.getRegionServers(true).size());` should have made sure all RS are fine now. I understand in case that is not enough, you explicitly make sure the mini cluster is ready for other test cases. OK, good.

Reply:
> Why wait for decommission this server in the end?

Actually, we are waiting for server to be recommissioned (maybe method name not correct, last arg determines). Let me change the order.